### DISCUSSION TITLE
Unused Entities - filter out incompatible cards when adding

### DISCUSSION THREAD

**Discussion by SeanPM5** [home-assistant/frontend#11209] [Other feature requests]
Currently the Unused Entities flow does not take into account from which domain an entity belongs to, and does not perform any filtering or recommendations on the "add new card" screen. 

For example, if the user selects a `device_tracker` entity on the Unused Entities screen, and then wants to add to their UI, the "Add new card" dialog should either disable or filter out all cards that a device tracker cannot be added to - gauge, iframe, light, markdown, media control, plant, thermostat, shopping list, weather forecast, etc. 

![add-new-card-filtered-list](https://i.imgur.com/6LqcXgA.png)

Right now, it shows every available card type and just gives the user an error when they try to add it to an incompatible card. So the user ends up with a non-functional card and has to cancel and restart the process. This could be a little frustrating for beginners that haven't familiarized themselves with all two dozen card types yet.

Most of these cards only accept a single domain (alarm panel, gauge, light, media, plant, sensor, thermostat, weather). So for example if the user selects entities on Unused Entities screen and none of them are from the `light` domain, scratch the light card off the list of available cards.

And some of these cards such as iframe, markdown, and shopping list _never_ accept any entities period, so there's no reason to show them when in the Unused Entities flow.

Showing only compatible cards would make this foolproof for beginners, and easier for everyone. Could make quicker decisions on what card to add something to when you're only seeing relevant ones.

**Comment by bramkragten** [comment]
In 0.104 things changed a bit, in that we do suggest a card to the user based on the entities, and have an option to still choose your own config (and then this still applies)

![image](https://user-images.githubusercontent.com/5662298/71834527-b6a68d80-30af-11ea-9d09-b427b1174dac.png)

**Comment by SeanPM5** [comment]
Dang that is slick! Nice work 😃

**Comment by spacegaier** [comment]
@SeanPM5 So can this be closed? There of course can be situations where multiple different card types could potentially be used for a given entity, but since the frontend now proposes a valid set, there should not be any errors anymore.

Or do you envision something more (e.g. marking for the user which entities have multiple options)? That might however be quite overwhelming when many unused entities are processed in bulk.

**Comment by SeanPM5** [comment]
The initial card suggestion prompt is helpful and does alleviate this to a certain extent, but the user might not always like what is suggested by default. You might prefer your lights to be inside a single Entities card rather than multiple Light cards for example. So I think a lot of people will still opt to "Pick a Different Card" and the issues I mentioned still apply to the card picker there.

I think the following three changes would greatly improve the user experience:

1. Filter out Shopping List, Web Page, Markdown cards from card picker when it's initiated from the Unused Entities flow. This flow is entirely about adding entities to your UI, and these three cards don't support entities. Why show cards that will _never_ work under any circumstances?
2. If the user has a **single entity selected**, compare the domain and filter out incompatible cards.
    - If I'm just trying to add my `vacuum.roborock_s6` entity to the UI, I shouldn't be shown tons of incompatible cards like Alarm Panel, Calendar, Humidifier, Light, Map, Markdown, Media Control, Picture cards, Plant, Thermostat, Webpage, Shopping List, Weather, etc.
3. If the user has **multiple entities selected**, only show cards which are actually capable of supporting multiple entities (this includes Glance card, Entities card, Logbook card, etc). 

I don't think there would need to be a ton of bulk processing or fancy logic involved, although it's possible I might be missing some edge case.
